### ISSUE TITLE
[FEATURE] Introduce Percentage-based Targeting for In-App Surveys

### ISSUE THREAD

**Issue by jonas-hoebenreich** [formbricks/formbricks#1960]
### Is your feature request related to a problem? Please describe.

It would be great to have a feature that allows the targeted display of in-app surveys to a specific percentage of users. This functionality, inspired by platforms like Hotjar, would provide more nuanced control over survey distribution.

### Describe the solution you'd like

The envisioned solution involves assigning each user to one of several buckets (e.g., 1-10), and allowing administrators to specify the buckets where the survey should run. This way, precise targeting, such as reaching only 10% of the user base, can be achieved effortlessly.

Our ideal implementation would include the following elements:
1. Users are distributed across buckets (1-10) upon widget initialization.
2. Admins can selectively choose which buckets the survey should be displayed to, enabling percentage-based targeting.
3. Flexibility to assign users to specific buckets during widget initialization, providing granular control over survey exposure.

**Comment by jobenjada** [comment]
Hey Jonas! This sounds interesting :) A few follow up questions:

1. Will the users remain in the buckets for later targeting or follow up questions? Or do you just want reach e.g. every 10th visitor? This could be achieved much easier I suppose.

2. In point 3 you request assigning users to specific buckets. Based on which attributes or criteria would you want to do that? It seems like we will provide a lot of the more granular targeting. you are requesting once this is merged:

https://github.com/formbricks/formbricks/pull/758

Cheers!

**Comment by jonas-hoebenreich** [comment]
Hi Johannes,

1. What do you mean by "visitor"? Pageview, session, user? But I guess as long as you can target how many users see your survey its fine.
2. Regarding issue #758, is my understanding correct that it involves the utilization of backend attributes (that will be almost impossible to scale)? I'm envisioning a functionality where we could fire something like formbricks.setFrontendAttribute("bucket", "1") that will be saved in-memory during a pageview. This would allow us to define attributes on the frontend exclusively, without the need for synchronization with the backend. In the backend there should be an option like "display survey when frontend attribute XYZ is equal to ABC" Such an enhancement would enable us to selectively target users, for example, triggering surveys exclusively for users in bucket 1. Additionally, these frontend attributes could be added to responses, serving as additional metadata.

**Comment by jobenjada** [comment]
Hey Jonas!

**1. Randomizer**
We decided ship a randomizer soon, as this is quite easy to build (e.g. only survey 10% of people who trigger a survey). @Dhruwang will be on it next week 🤸 

**2. Client-side filtering**
As for client-side attributes we will postpone shipping this until we have the Advanced User Targeting for identified user merged (likely next week). We're planning to update our Survey Editor to make the distinction between Website Surveys (unidentified visitor) and In App Surveys (identified) more clear. This gives us room to ship the client-side filtering as well as other features tools like HotJar Ask provide without cluttering the UX for companies using Formbricks with identified users. In the process of this Survey Editor update, we'll ship the client-side filtering 🚢 

I'll likely follow up with a few quesitons once we have thought it through a bit better 🤓

**Comment by jobenjada** [comment]
@jonas-hoebenreich The Randomizer is shipped, does this help?

**Comment by jonas-hoebenreich** [comment]
Yes, thank you so much 😊 
I think it makes sense to close this issue and create a new dedicated issue for frontend attributes
